modules/swagger-codegen/src/main/resources/lua/model.mustache

```diff
@@ -13,9 +13,9 @@ local function cast_{{classname}}(t)
 end
 
 local function new_{{classname}}({{#vars}}{{name}}{{#hasMore}}, {{/hasMore}}{{/vars}})
-  return cast_pet({
+  return cast_{{classname}}({
     {{#vars}}
-    {{name}} = {{name}};
+    ["{{baseName}}"] = {{name}};
```

Comment:
Also, it would be better to prefer the non-quoted form (saving the `[]` syntax only for invalid lua identifiers)